### ISSUE TITLE
実装: non-playing terminal reward v3 を導入して競り参加率を再調整

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#232]
Parent: #202
Depends on: #230
Base branch: `feature/issue-202-non-playing-ai`

## 背景

Reward v2（Napoleon win=`3*d`, lose=`-5`, Adjutant win=`d`, Citizen win=`d`）を導入した30 iteration実験では、PASS崩壊は解消した一方で、今度は競り参加が強すぎる方向へ振れた。

Reward v2 run:
`~/napoleon_runs/non-playing-ppo-reward-v2`

iter30:
- PASS rate: 0.0249
- candidate Napoleon formation rate: 0.982
- all-pass forced d=12: 8
- declaration success rate: 0.2831
- mean reward: 7.632

これにより、PPOはreward設計に反応して競り方策を変えられることが確認できた。

次はrewardをより自然な水準へ戻し、競り参加率が極端にならないかを再検証する。

## 目的

non-playing terminal rewardをv3へ変更する。

宣言枚数を `d` とする。

### 勝利時

- Napoleon: `2 * d`
- Adjutant: `d`
- Citizen: `d`
- Napoleon + Adjutant: `3 * d`

### 敗北時

- Napoleon: `-5`
- Adjutant: `0`
- Citizen: `0`
- Napoleon + Adjutant: `-5`

Napoleon+AdjutantはNapoleonとAdjutantの勝利報酬を加算した `3*d` とする。

## 狙い

Reward v2の `3*d` はPASS崩壊を解消したが、ほぼ常時Napoleonを取りに行く方策になった。

Reward v3ではNapoleon勝利報酬を `2*d` に下げ、reward v1のPASS偏重とreward v2の競り参加偏重の中間に落ち着くかを確認する。

このIssueではreward以外の条件は変更しない。

## 実装方針

- reward version / id をv3へ更新する
- bidding / adjutant / exchangeのterminal reward計算をReward v3へ統一する
- manifest / checkpoint / run config / provenanceに新しいreward versionを記録する
- reward v1/v2 runをv3で誤resumeしないようfail-closeする
- rollout topology / opponent mix / PPO設定は維持する

## 維持する条件

- candidate x1 + frozen x4
- candidate 5-seat rotation
- Frozen bidding opponents: conservative/passive 50:50 per-seat seeded mix
- playing: `ppo-separated-v1000` frozen
- PPO hyperparametersは変更しない
- all-pass forced d=12のゲームルールは変更しない

## テスト

最低限、固定fixtureでrole別rewardを確認する。

### d=13
- Napoleon win = 26
- Napoleon lose = -5
- Adjutant win = 13
- Adjutant lose = 0
- Citizen win = 13
- Citizen lose = 0
- Napoleon+Adjutant win = 39
- Napoleon+Adjutant lose = -5

### d=19
- Napoleon win = 38
- Adjutant win = 19
- Citizen win = 19
- Napoleon+Adjutant win = 57

all-pass forced contractでは従来どおり `d=12` を用いてReward v3を計算する。

## 30 iteration再実験

実装後、Reward v2 runと同条件で新規runを実施する。

```text
iterations             30
games-per-iteration    200 logical seeds
epochs                  4
batch-size              128
learning-rate           1e-4
evaluation-interval     5
evaluation-games        100
seed                    202
```

新しいoutput directory例:
`~/napoleon_runs/non-playing-ppo-reward-v3`

旧runは上書きしない。

## 比較指標

iter 5 / 10 / 15 / 20 / 25 / 30について最低限以下を比較する。

- candidateBiddingDecisionCount
- passRate
- bidCount
- candidateNapoleonFormationRate
- declarationSuccessRate
- allPassForcedD12Count
- meanReward
- targetPointCardsDistribution 13〜19
- fallbackCount
- illegalActionCount

主比較対象:
- reward v1 mixed frozen: `~/napoleon_runs/non-playing-ppo-mixed-frozen-v1`
- reward v2: `~/napoleon_runs/non-playing-ppo-reward-v2`

## Acceptance Criteria

- Reward v3が上記定義で実装される
- Napoleon+Adjutant winが `3*d` になる
- reward version/idが更新され、artifact provenanceに保存される
- reward v1/v2 runをreward v3で誤resumeしない
- candidate x1 + frozen x4 + rotation + mixed frozen bidding topologyを維持する
- fixed playing policyを維持する
- role別rewardのunit testが通る
- 2〜3 iteration smokeが通る
- fallback=0 / illegal action=0
- 実装完了時に30 iteration比較実験用のコピペ可能なコマンドを提示する

## 非目標

- PPOアルゴリズム変更
- learning rate / clip / batch size等のhyperparameter tuning
- Frozen opponent mix変更
- all-pass forced d=12ルール変更
- playing policy再学習
- 親PR #203 のmainへのMerge

## PR運用

現在の `napoleon-web-issue-202` 作業ディレクトリをそのまま使用し、新しいworktreeは作らない。

`feature/issue-202-non-playing-ai` を親としてIssue専用ブランチを作成する。
PR baseは `feature/issue-202-non-playing-ai` とする。

Codex review通過・必要テスト通過後はReady for reviewにし、そのまま親ブランチへmergeする。`main`にはmergeしない。
